Networking/NetworkingTests/Mapper/ProductVariationMapperTests.swift

```diff
@@ -27,6 +27,15 @@ final class ProductVariationMapperTests: XCTestCase {
 
         XCTAssertEqual(productVariation, sampleProductVariation(siteID: dummySiteID, productID: dummyProductID, id: 2783))
     }
+
+    /// Verifies that the fields of the Product Variations with alternative types are parsed correctly when they have different types than in the struct.
+    /// Currently, `price`, `regularPrice`, `salePrice`, `manageStock`, `purchasable`, and `permalink`  allow alternative types.
+    ///
+    func test_that_product_variations_alternative_types_are_properly_parsed() throws {
+        let productVariation = try XCTUnwrap(mapLoadProductVariationResponseWithAlternativeTypes())
+
+        XCTAssertEqual(productVariation, sampleProductVariationAlternativeTypes(siteID: dummySiteID, productID: dummyProductID, id: 1275))
```

Comment:
Nit: instead of comparing the whole object, how about asserting each of the properties with the alternative types to ensure we get the expected results? IMO it would be clearer to see the test results.